### PR TITLE
Added regex to remove external link

### PR DESCRIPTION
**Pull request by sladyn98**

### CODE REVIEW

**Comment by halkeye** [comment]
Will that regex work for both asciidoc and markdown?

**Comment by halkeye** [comment]
i'll have to dig into it more, but you can probaly write a test with convertBody with adoc and md and confirm both outputs have it, then run your code (pulled into a new function so it can be tested)

```
it('removes external link', async () => {
const body = await convertBody(..., 'adoc');
expect(cleanupExternalLinks(body)).toBe('foo');
});
```

or something

File: utils.test.js [hunk 1]
```diff
@@ -31,5 +31,8 @@ describe('utils', function() {
   it('checkUrl', () => {
     expect(checkUrl(['wiki.jenkins.io'], 'https://wiki.jenkins.io/something')).toBe(true);
   });
+  it('removeExternalLink', () => {
```
Comment (sladyn98) [inline utils.test.js]: Added a basic test, is there a way I can run and check if this works?

Comment (halkeye) [reply utils.test.js]: `npm run test` should do it.

File: index.js [hunk 2]
```diff
@@ -127,7 +127,8 @@ async function processContent(req, res, wikiContent, extension, archiveFormat) {
     const files = [];
     const images = findImages(wikiContent).map(decodeEntities);
     const urlRE = new RegExp('(' + images.map((i) => i.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&')).join('|') + ')', 'gi');
-    const content = await replaceAsync(stdout, urlRE, async function(val, grab) {
+    const parsedURL = urlRE.replace(/{.external-link}./, "");
```
Comment (halkeye) [inline index.js]: Can you use your newly created function here?

Comment (sladyn98) [reply index.js]: Yeah sure

File: utils.test.js (hunk 1 above)
Comment (sladyn98) [reply utils.test.js]: Thanks

File: utils.test.js [hunk 3]
```diff
@@ -31,5 +32,8 @@ describe('utils', function() {
...
+    expect(removeExternalLink("(https://github.com/jenkinsci/plugin-pom){.external-link}.")).toBe("(https://github.com/jenkinsci/plugin-pom)");
```
Comment (sladyn98) [inline utils.test.js]: @halkeye I am not too sure how to check for adoc and MD, is there any resource I can refer to.

Comment (halkeye) [reply utils.test.js]: Sorry I'm a bit distracted by a few others projects, being sick, and work thing

But i'm paying attention I swear :)

I don't think anything comes to mind. Maybe find a plugin where this is happening, and export both an ascii and a markdown file, then add those both as strings/files for the tests?

Comment (sladyn98) [reply utils.test.js]: I will try and give it a shot

**Comment by sladyn98** [comment]
@timja This should solve it

File: utils.test.js (hunk 3 above)
Comment (sladyn98) [reply utils.test.js]: Yeah I added an markdown file for test purposes

File: utils.js [hunk 4]
```diff
@@ -225,9 +225,20 @@ function checkUrl(validWikiDomains, url) {
...
 }
 
+/**
+ * Removes external links from urls
+ * @param {string} url to check
+ * @return {string} parsed string
+ */
+function removeExternalLink(url) {
+  const regex = /{.external-link}.|{.external-link}|{.external-link}\\/g;
+  return url.replace(regex, "");
```
Comment (timja) [inline utils.js]: just tested this locally, got:
```
Error: Error: url.replace is not a function
```

plugin used was `chatwork`

**Comment by timja** [comment]
@sladyn98 alternative implementation here: https://github.com/jenkins-infra/jenkins-wiki-exporter/pull/55 no tests yet though

**Comment by timja** [comment]
I've added a test now, I think this can be closed in favour of: https://github.com/jenkins-infra/jenkins-wiki-exporter/pull/55

**Comment by halkeye** [comment]
so we merged #55 instead, so i'll close this

**Comment by sladyn98** [comment]
Cool. Still learning JS ill probably be able to make a better PR next time :)

**Pull request closed without merging**